hackvote/app/controllers/events_controller.rb

```diff
@@ -0,0 +1,41 @@
+class EventsController < ApplicationController
+
+  NUMBER_OF_VOTES = 3
+
+  def new
+	  @event = Event.new
+	  @cur_events = Event.where( completed: false)
+	  @past_events = Event.where( completed: true)
```

Comment:
So right now the `events` table isn't indexed on `completed`, so both of these queries will have to iterate over all the rows to check if they're completed or not. So 2 passes over the entire data set. Whereas if you had all the data, you could use the [`partition`](http://ruby-doc.org/core-2.2.2/Enumerable.html#method-i-partition) method from `Enumerable` to pass over the data once.

There's also overhead in going out across the network to get the results of the query.

If `completed` was indexed in SQL, it would probably be easiest + faster than partitioning to have 2 separate SQL queries. I probably should have recommended you index on `completed` if you're doing queries based on it.